File: leetcode-submissions/0167-two-sum-ii---input-array-is-sorted/solution.java
class Solution {


    public int[] twoSum(int[] nums, int target) {
        int lo = 0; int hi = nums.length - 1;
        int[] ans = new int[2];

        while (lo < hi) {
            int sum  = nums[lo] + nums[hi];
            if (sum < target) {
                lo++;
            } else if (sum > target) {
                hi--;
            } else {
                ans[0] = lo + 1;
                ans[1] = hi + 1;
                return ans;
            }
        }
        return ans;
    }
}

File: leetcode-submissions/0167-two-sum-ii---input-array-is-sorted/solution.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        #we will use two pointers to solve this problem
        #we set our left and right pointer
        l, r = 0, len(numbers) - 1
        #we iterate through our list
        while l < r:
            tot = numbers[l] + numbers[r]
            if tot > target:
                r -= 1
            elif tot < target:
                l += 1
            else:
                return [l+1, r+1]
